src/test/java/com/cappella/csv/CsvClientTests.java

```diff
@@ -0,0 +1,165 @@
+package com.cappella.csv;
+
+import java.beans.Transient;
+import java.io.InputStream;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Locale;
+import java.util.HashMap;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.commons.csv.CSVRecord;
+import com.cappella.model.Problems;
+import com.cappella.model.TaskData;
+import com.cappella.model.SubTask;
+
+class CsvClientTests {
+
+        final Logger LOGGER = LoggerFactory.getLogger(getClass());
+
+        @Test
+        void testNoHeadersCsvFile() {
+                // the simplest way to malform the header is to add a trailing comma
+                ClassLoader classloader = this.getClass().getClassLoader();
+                InputStream is = classloader.getResourceAsStream("noHeaders.csv");
+                Problems problems = new Problems();
+                CsvClient csv = new CsvClient();
+                List<TaskData> grantTasks = csv.parseGrantCsvToTasks(is, problems);
+                Assertions.assertTrue(problems.getErrors().containsKey(Problems.ERROR_PARSING_CSV_FILE));
+        }
+
+        @Test
+        void testMalformedHeadersCsvFile() {
+                // the simplest way to malform the header is to add a trailing comma
+                ClassLoader classloader = this.getClass().getClassLoader();
+                InputStream is = classloader.getResourceAsStream("malformedHeader.csv");
+                Problems problems = new Problems();
+                CsvClient csv = new CsvClient();
+                List<TaskData> grantTasks = csv.parseGrantCsvToTasks(is, problems);
+                Assertions.assertTrue(problems.getErrors().containsKey(Problems.ERROR_PARSING_CSV_FILE));
+        }
+
+        @Test
+        void testMalformedDataCsvFile() {
+        }
+
+        @Test
+        void testMissingNameHeaderCsvFile() {
+                ClassLoader classloader = this.getClass().getClassLoader();
```

Comment:
The first five lines of code in this tests and a bunch of tests below could be collapsed into a single private method that accepts a file name as a parameter.

Reply:
fixed in updated PR